src/InteractiveWindow/Editor/InteractiveWindow.UIThreadOnly.cs

```diff
@@ -1386,6 +1386,8 @@ private void AddLanguageBuffer()
                     bufferAdded(_window, new SubmissionBufferAddedEventArgs(buffer));
                 }
 
+                _window.LanguageBufferCounter++;
+
```

Comment:
Are we counting everytime AddLanguageBuffer method is called? Shouldn't this counter be updated in that method instead?